### ISSUE TITLE
/cancel endpoint

### ISSUE THREAD

**Issue by ianweller** [fedora-infra/datagrepper#71]
Cancel a submitted job. /cancel?id=<int>

Checks if you are logged in as the same user who submitted the job. Can only cancel a job if it is free (open jobs are already being processed). Sets job state to "canceled".

Job runner should check before setting each job to open whether or not it has been canceled.

**Comment by ralphbean** [comment]
No longer needed after #158.
